libminifi/src/core/state/nodes/SupportedOperations.cpp

```diff
@@ -64,9 +65,12 @@ void SupportedOperations::addProperty(SerializedResponseNode& properties, const
 
 SupportedOperations::Metadata SupportedOperations::buildUpdatePropertiesMetadata() const {
   std::vector<std::unordered_map<std::string, std::string>> supported_config_updates;
-  for (const auto& config_property : minifi::Configuration::CONFIGURATION_PROPERTIES) {
-    if (!update_policy_controller_ ||
-        (update_policy_controller_ && update_policy_controller_->canUpdate(std::string(config_property.name)))) {
+  for (const auto& config_property : Configuration::CONFIGURATION_PROPERTIES) {
+    auto sensitive_properties = Configuration::getSensitiveProperties(configuration_reader_);
+    if (ranges::find(sensitive_properties, config_property.name) != ranges::end(sensitive_properties)) {
+      continue;
+    }
+    if (!update_policy_controller_ || (update_policy_controller_ && update_policy_controller_->canUpdate(std::string(config_property.name)))) {
```

Comment:
```suggestion
    if (!update_policy_controller_ || update_policy_controller_->canUpdate(std::string(config_property.name))) {
```

The second checking of update_policy_controller_ seems unnecessary.

Reply:
Good point, updated in 8561e67e7453596a5158f2ab937fc724ec551974